nixos/modules/services/misc/home-assistant.nix

```diff
@@ -5,7 +5,7 @@ with lib;
 let
   cfg = config.services.home-assistant;
 
-  configFile = pkgs.writeText "configuration.yaml" (builtins.toJSON cfg.config);
+  configFile = pkgs.writeText "configuration.json" (builtins.toJSON (lib.recursiveUpdate defaultConfig cfg.config));
```

Comment:
What does `recursiveUpdate` do if `cfg.config == null`?

Reply:
Good catch - short answer: "I don't know" so now we check for it being null.